libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/internal/location/ConstantVelocityInterpolatorTest.kt

```diff
@@ -0,0 +1,59 @@
+package com.mapbox.navigation.ui.maps.internal.location
+
+import com.mapbox.geojson.Point
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import kotlin.math.pow
+
+@RunWith(RobolectricTestRunner::class)
+class ConstantVelocityInterpolatorTest {
+
+    @Test
+    fun `should calculate timings for constant velocity`() {
+        val p0 = Point.fromLngLat(0.0, 0.0)
+        val keyPoints = arrayOf(
+            Point.fromLngLat(0.3, 0.4),
+            Point.fromLngLat(0.6, 0.8),
+            Point.fromLngLat(1.0, 1.0),
+        )
+
+        val sut = ConstantVelocityInterpolator(p0, keyPoints)
+
+        val path = calculatePathValues(p0, *keyPoints)
+        assertEquals(keyPoints.size, path.size)
+        path.forEach { (outTime, inTime) ->
+            assertEquals(outTime, sut.getInterpolation(inTime))
+        }
+    }
+
+    private fun calculatePathValues(p0: Point, vararg points: Point): List<Pair<Float, Float>> {
```

Comment:
Is it just the copypaste of ConstantVelocityInterpolator init? Is there really much sense in such a test? I'd say we should take a simple example, calculate the expected value manually and check it in the test.

Reply:
This test not only validates PathInterpolator values but also verifies velocity is constant. Calculating this by hand and hardcoding the result here brings no value.

Reply:
Why no value? To be more precise, why does it bring less value than copypasting the code what you test? If you've made an error in the implementation it will also be present here. What's the point of the test then?

> This test not only validates PathInterpolator values but also verifies velocity is constant.

If this causes some issues it can be divided into different tests.

Reply:
I know it can, but there is no need for it. `calculatePathValues` method already does that for us. It performs a self-check to guarantee that returned "verfication path" does produce timing for constant velocity.
With that being guaranteed, we can use that "verification path" to validate  ConstantVelocityInterpolator values.